### DISCUSSION TITLE
Is it possible to read the layers like the PDFTron library does?

### DISCUSSION THREAD

**Discussion by ederjuarez** [wojtekmaj/react-pdf#965] [Q&A]
We currently use PDFTron in development to read the layers of the PDF you generate in AutoCad, is it possible with this library?

**Comment by wojtekmaj** [comment]
In React-PDF itself? No. However, React-PDF gives you an easy way to hook into `document` and `page` objects in `Document`'s `onLoadSuccess` and `Page`'s `onLoadSuccess` respectively, and from there, you have full access to PDF.js API.

Whether PDF.js API allows that - that would be the question for PDF.js developers. I don't see any method name that would ring the bell, but perhaps you could start your research with `page.getStructTree()`?
